### DISCUSSION TITLE
working with bitwarden

### DISCUSSION THREAD

**Discussion by robina80** [authelia/authelia#4072] [Q&A] [answered]
hi all,

is there a way i can get authelia working with bitwarden, so authelia handles the logins/2fa for bitwarden

thanks,
rob

**Comment by robina80** [comment]
bumping this thread

**Comment by james-d-elliott** [comment, marked as answer]
There is not really a free way to do this. Bitwarden Enterprise supports SSO, but it requires the user enter the password anyway since the password is used for the HMAC secret to decrypt the vaults. In addition the web vault from memory uses fragments (parts of the URL after a `#`) which are invisible to proxies and thus invisible to Authelia (they exist only in the client, only visible to ECMAScript/javascript).
